### PR TITLE
Update vendoring to match moby/moby 

### PR DESCRIPTION
**Pull request by euanh**
Update vendored dependencies to match the versions used by moby/moby

This change moves libnetwork's version of gocapability ahead of moby's, so moby will also have to be updated.

This pull request supersedes #2028

### CODE REVIEW

**Comment by euanh** [comment]
I'll squash the vendor.conf changes before merging.   I want to keep them separate until CI gives the green light.

**Comment by selansen** [comment]
LGTM

**Comment by fcrisciani** [comment]
I see you are touching the Dockerfile, I would like to prioritize the circle ci migration PR before this

**Comment by euanh** [comment]
@fcrisciani I'll move the Dockerfile change into another pull request (although it's just bumping the Go micro version).   It would be good to get the rest of the changes in as soon as possible, before we start to fall behind moby/moby again.

**Comment by fcrisciani** [comment]
@euanh can you rebase on latest master? so to have a run on the new CI infra?

File: vendor.conf [hunk 1]
```diff
@@ -1,27 +1,28 @@
...
+github.com/docker/docker 162ba6016def672690ee4a1f3978368853a1e149
+github.com/docker/go-connections 7beb39f0b969b075d1325fecb092faf27fd357b6
 github.com/docker/go-events 9461782956ad83b30282bf90e31fa6a70c255ba9
 github.com/docker/go-units 9e638d38cf6977a37a8ea0078f3ee75a7cdb2dd1
 github.com/docker/libkv 1d8431073ae03cdaedb198a89722f3aab6d418ef
 
 github.com/godbus/dbus v4.0.0
-github.com/gogo/protobuf v0.4
-github.com/golang/protobuf 7a211bcf3bce0e3f1d74f9894916e6f116ae83b4
+github.com/gogo/protobuf v1.0.0
```
Comment (fcrisciani) [inline vendor.conf]: I guess we should regenerate the protobuf if we update the gogo library

**Comment by fcrisciani** [comment]
@euanh can you also bring in the script or an automated way to do this operation next time?

**Comment by euanh** [comment]
@fcrisciani I used a bit of sed and awk to update the packages which were common between moby and libnetwork, but after that there was a bit of manual fiddling.   I'm not sure it can be made fully automatic.

I'm planning to make a separate pull request to add an updater.   I don't want this PR to get bogged down - that's why #2028 was never merged.

**Pull request merged**